go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -204,89 +199,68 @@ func transformRoutePlan(ctx *plancontext.PlanningContext, op *operators.Route) (
 }
 
 func transformUpdatePlan(ctx *plancontext.PlanningContext, op *operators.Route, upd *operators.Update) (logicalPlan, error) {
-	var vindex vindexes.Vindex
-	var values []evalengine.Expr
-	if op.Selected != nil {
-		vindex = op.Selected.FoundVindex
-		values = op.Selected.Values
-	}
 	ast := upd.AST
 	replaceSubQuery(ctx, ast)
+	rp := &engine.RoutingParameters{
+		Opcode: op.Routing.OpCode(),
+	}
+	op.Routing.UpdateRoutingParams(ctx, rp)
```

Comment:
I have the same question here. Why are we not assigning the keyspace even though we previously used to set the keyspace on the `RoutingParameters`:
```go
RoutingParameters: &engine.RoutingParameters{
			Opcode:            op.RouteOpCode,
			Keyspace:          op.Keyspace,
			Vindex:            vindex,
			Values:            values,
			TargetDestination: op.TargetDestination,
		},
```

Reply:
Same answer here :)

The difference here is that for UPDATE/DELETE we require a keyspace, so the default one is not needed.